src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -44,27 +45,15 @@ class TransposeValueTensors : public ov::pass::MatcherPass {
         // NB: Transpose Parameter that correspond to V-tensor it will
         // speed-up its multiplication with attention scores
         std::swap(param_shape[2], param_shape[3]);
-        auto new_param = std::make_shared<ov::opset13::Parameter>(matched_param->get_element_type(), param_shape);
-        new_param->set_friendly_name(matched_param->get_friendly_name());
-        new_param->outputs().begin()->get_tensor().set_names(
-            matched_param->outputs().begin()->get_tensor().get_names());
-        ov::replace_node(matched_param, new_param);
-        // NB: Save in order to add/remove to the model later on
-        ctx.get().new_params.push_back(new_param);
-        ctx.get().old_params.push_back(matched_param);
 
-        auto order_cst = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{4}, {0, 2, 3, 1});
-        auto new_transpose =
-            std::make_shared<ov::opset13::Transpose>(matched_transpose->input_value(0), order_cst->output(0));
-        new_transpose->set_friendly_name(matched_transpose->get_friendly_name());
-        ov::replace_node(matched_transpose, new_transpose);
+        matched_param->set_partial_shape(param_shape);
 
-        auto new_concat =
-            std::make_shared<ov::opset13::Concat>(ov::OutputVector{new_param->output(0), new_transpose->output(0)}, 3u);
-        new_concat->set_friendly_name(matched_concat->get_friendly_name());
-        ov::replace_node(matched_concat, new_concat);
+        auto order_cst = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{4}, {0, 2, 3, 1});
 
+        matched_transpose->set_argument(1, order_cst);
+        matched_concat->set_axis(3u);
```

Comment:
That's great if it modification of existing node works. Previously it wasn't the case by some reason...